arrow-array/src/cast.rs

```diff
@@ -21,6 +21,11 @@ use crate::array::*;
 use crate::types::*;
 use arrow_data::ArrayData;
 
+#[doc(hidden)]
+pub mod __private {
```

Comment:
Some context might be nice:

```suggestion
/// Re-export symbols needed for downcast macros
/// 
/// Name follows `serde` convention
#[doc(hidden)]
pub mod __private {
```